### ISSUE TITLE
WooCommerce Archive-product.php Override

### ISSUE THREAD

**Issue by aaronditto** [understrap/understrap#373]
I'm attempting to style the WooCommerce shop page, which is traditionally archive-product.php.

However, when I copy over the archive-product.php file from the WooCommerce plugin folder (templates\archive-product.php) and put it in my Child Theme's woocommerce folder - once I start making edits to the file, they are not being displayed.

I see in understrap\woocommerce.php that it allows for detecting this override, but I can't seem to figure out why it's not picking up this file, and continues to default to just using woocommerce_content( ); loop.

Also - doesn't the $default_path variable in woocommerce.php point to "plugin_dir_path" \woocommerce - as in the Plugin directory path when looking for archive-product.php - and instead - should be looking for it under childtheme\woocommerce\archive-product.php

Do I need to make any modifications to woocommerce.php to achieve what I'm wanting? What else am I missing? :-)

**Comment by Thomas-A-Reinert** [comment]
Just wanted to mention that it works like a charm on my installations with Understrap child themes, so there may be something wrong with your installation?

archive-product.php is in the woocommerce subfolder? sure there are no typos?

**Comment by aaronditto** [comment]
No typos - I literally copy the archive-product.php file from the WooCommerce\Templates plugin folder and place it into childtheme\woocommerce

No matter what changes I make, they do not display. I've even tried a completely empty file called archive-product.php and it still pulls the products without any issue. I would assume if the override was working - and it pulled a blank archive-product.php file - that it wouldn't display any products at all.

**Comment by Thomas-A-Reinert** [comment]
Hm, maybe you got caching active?
Try deactivating any caching plugin you may have active.
Try opening an incognito window and keep dev-tools open. Try to force reload too or empty caches. Remember I had some problems with that too and felt like a dork at the end :/

**Comment by aaronditto** [comment]
No caching plugins installed - still on local dev environment.

Cleared all caches, incognito, etc - no change.

Interestingly enough, even with archive-product.php in childtheme\woocommerce - I can open up woocommerce.php in the childtheme folder, and delete line #48 which calls the woocommerce_content( ); in teh if/else statement - and if that is deleted, nothing populates on the page. This would indicate that it's not detecting the archive-product.php file in the woocommerce folder.

Thanks for your help & time on this. I'm starting to pull my (gray) hair out. ;-)

**Comment by Thomas-A-Reinert** [comment]
Hm, that´s strange in fact. I know I also did some changes to my theme and had some hustle to set it up working with woo correctly but cannot recall which. :/

If you want, I can give you access to my child-theme and you could make a diff. If yu´d find the solution/reasons why, I guess that would be a great help to all using UnderStrap!

But, be warned: That thing is _heavily_ customized and still work in progress, so some things may not be solved elegantly..

**Comment by ZacharyElkins** [comment]
From WC Website:

> Gotcha: If your theme has a woocommerce.php file, you will be unable to override the woocommerce/archive-product.php custom template in your theme, as woocommerce.php has priority over other template files. This is intended to prevent display issues.

> 

If you look at this commit, they reverted to using woocommerce.php instead of multiple template files in Understrap:
https://github.com/holger1411/understrap/commit/80ff0f07fd3c1217537469940454b8c6e47d9411

**Comment by aaronditto** [comment]
Thanks Zachary - That makes complete sense and that's what I thought too.

I found that commit earlier today as well - but what I don't understand - is if you download the latest version from Github today - it includes woocommerce.php with this part of code instead of what is shown in that committ

`<?php 
				$template_name = '\archive-product.php'; 
				$args = array(); 
				$template_path = ''; 
				$default_path = untrailingslashit( plugin_dir_path(__FILE__) ) . '\woocommerce';

					if ( is_singular( 'product' ) ) {

						woocommerce_content();

			//For ANY product archive, Product taxonomy, product search or /shop landing page etc Fetch the template override;
				} 	elseif ( file_exists( $default_path . $template_name ) )
					{
					wc_get_template( $template_name, $args, $template_path, $default_path );

			//If no archive-product.php template exists, default to catchall;
				}	else  {
					woocommerce_content( );
				}

			;?>`

So what am I missing? LOL

**Comment by aaronditto** [comment]
@ZacharyElkins Looks like they added it back in this commit

https://github.com/holger1411/understrap/commit/0b968fe9c37543a8fb7eb9f07eadc6ae957400a8

@holger1411 Any chance you could weigh in on this? :-)

**Comment by typeplus** [comment]
I added code previously that looked for an archive-product.php a few releases ago but this commit was removed. So it then just had the catchall applied and it ignored any archive-product.php template you have.

I added this new code to basically work if you have a template override and still work with the catchall if you didn't.  Since you are using the child theme I think you just need to copy over the woocommerce.php into the root of the child theme and it will make sure your archive-product.php template overrides are used. I don't use child themes but tested it and it worked.

**Comment by ChrisLTD** [comment]
I ran into this issue myself, the issue was that on my local machine the slashes were mixed. My machine uses forward slashes, but `$template_name` and `$default_path` have backslashes hardcoded:

```php
$template_name = '\archive-product.php'; 
$args = array(); 
$template_path = ''; 
$default_path = untrailingslashit( plugin_dir_path(__FILE__) ) . '\woocommerce';
```

**Comment by yamaks** [comment]
how did you solve this problem with the custom template?
I put my template in the theme woocommerce folder and it does not work 
themes/my theme/woocommerce/archive-product.php

**Comment by yamaks** [comment]
Can i just remove woocommerce.php from my theme folder?
If I delete this file, then the archive-product.php is overridden

**Comment by holger1411** [comment]
Closing this because: Will be fixed in 0.8.2.
We replaced WooCommerce.php and hook the woo wrapper into the markup via inc/woocommerce.php
